taps/serializers.py

```diff
@@ -16,6 +16,8 @@ class TapSerializer(serializers.ModelSerializer):
         min_value=0, max_value=100, allow_null=True, required=False,
     )
     room = RoomSerializer(read_only=True)
+    time_added = serializers.DateTimeField()
```

Comment:
Make these read-only as well since users should never update them.

Then, create a trivial `update()` method that updates time_updated before saving:

```python
def update(self, instance, validated_data):
    instance.time_updated = timezone.now()
    return super().update(instance, validated_data)
```

Reply:
I don't believe that's entirely true, many of the providers have added/updated which is different from the time added/updated in our db.

Reply:
Right. This is for API users only, which your parsing code will not use.